### PR TITLE
Data Explorer: Request multiple column profiles at once, flesh out histogram and frequency table parameters, add simple implementations for pandas and polars

### PR DESCRIPTION
**Pull request by wesm**
This is work helping put the pieces in place for #4061. A few things here:

* get_column_profiles receives an array of profile specs (a "spec" is what I'm calling a profile type + its params), presumably no duplicates of the same time, and the results are returned in the ColumnProfileResult type
* Added simple parameters for histograms and frequency tables
* Add basic implementation and some unit tests for pandas. These will need to be fleshed out to support more cases and handle exceptional scenarios

This can't be merged in its current state until the corresponding Ark PR is landed but putting this up for review in the meantime. 

### QA Notes

There is nothing visible in the UI yet.

### CODE REVIEW

File: positron/comms/data_explorer-backend-openrpc.json [hunk 1]
```diff
@@ -1003,63 +1036,88 @@
...
-					"bin_width"
+					"num_bins"
 				],
 				"properties": {
-					"bin_sizes": {
+					"num_bins": {
+						"type": "integer",
+						"description": "Number of bins in the computed histogram"
+					},
+					"quantiles": {
```
Comment (dfalbel) [inline positron/comms/data_explorer-backend-openrpc.json:1050]: Should the computed `quantiles` be part of the output? Currently they seem to be unused.

File: positron/comms/data_explorer-backend-openrpc.json [hunk 2]
```diff
@@ -1003,63 +1036,88 @@
...
+						"description": "Counts of top values",
+						"items": {
+							"type": "integer"
+						}
 					},
-					"count": {
+					"other_count": {
 						"type": "integer",
-						"description": "Number of occurrences of value"
+						"description": "Number of other values not accounted for in counts. May be omitted"
```
Comment (dfalbel) [inline positron/comms/data_explorer-backend-openrpc.json:1120]: Maybe we want to document that this excludes `NAs`/ `Nulls`? To be consistent with the numeric histograms?

**Review by dfalbel — COMMENTED** [review]
The openRPC contract looks good to me

File: positron/comms/data_explorer-backend-openrpc.json (hunk 2 above)
Comment (wesm) [reply positron/comms/data_explorer-backend-openrpc.json]: done

**Comment by wesm** [comment]
I'll work on adding the polars implementation to this PR until we can coordinate with the corresponding Ark PR

**Comment by wesm** [comment]
Added the polars implementations

**Pull request merged**